test/controllers/concerns/authenticate/by_http_header_test.rb

```diff
@@ -0,0 +1,117 @@
+require 'test_helper'
```

Comment:
@tmaier I did a big refactor of auth & credentials because I'm about to add a few more methods. This current PR here attempted to change no logic, I simply broke things into separate files and moved the auth details from the User model into a dedicated HttpAuthCredential model (likewise for Password and Google).

You don't need to review this whole PR, but could you review this test file to sanity check things? I'll flag one specific question below

Reply:
Also, maybe you can sanity check this locally to make sure I didn't break things. I'm pretty confident everything still works because we had good test coverage before and I just moved code around, but I don't have this HTTP Header auth situation setup in my local dev so it's not easy for me to do a final test of it.